File: utils/benchmark/Phonebook/Phonebook.py

words=[
 u"James", u"John", u"Robert", u"Michael", u"William", u"David", u"Richard", u"Joseph",
 u"Charles", u"Thomas", u"Christopher", u"Daniel", u"Matthew", u"Donald", u"Anthony",
 u"Paul", u"Mark", u"George", u"Steven", u"Kenneth", u"Andrew", u"Edward", u"Brian",
 u"Joshua", u"Kevin", u"Ronald", u"Timothy", u"Jason", u"Jeffrey", u"Gary", u"Ryan",
 u"Nicholas", u"Eric", u"Stephen", u"Jacob", u"Larry", u"Frank", u"Jonathan", u"Scott",
 u"Justin", u"Raymond", u"Brandon", u"Gregory", u"Samuel", u"Patrick", u"Benjamin",
 u"Jack", u"Dennis", u"Jerry", u"Alexander", u"Tyler", u"Douglas", u"Henry", u"Peter",
 u"Walter", u"Aaron", u"Jose", u"Adam", u"Harold", u"Zachary", u"Nathan", u"Carl",
 u"Kyle", u"Arthur", u"Gerald", u"Lawrence", u"Roger", u"Albert", u"Keith", u"Jeremy",
 u"Terry", u"Joe", u"Sean", u"Willie", u"Jesse", u"Ralph", u"Billy", u"Austin", u"Bruce",
 u"Christian", u"Roy", u"Bryan", u"Eugene", u"Louis", u"Harry", u"Wayne", u"Ethan",
 u"Jordan", u"Russell", u"Alan", u"Philip", u"Randy", u"Juan", u"Howard", u"Vincent",
 u"Bobby", u"Dylan", u"Johnny", u"Phillip", u"Craig"]

# This is a phone book record.
class Record:
  def __init__(self, firstname, lastname):
    self.first = firstname
    self.last = lastname

  def __lt__(self, other):
    if self.last < other.last:
      return True
    if self.last > other.last:
      return False
    return self.first < other.first

Records = []

for first in words:
  for last in words:
    Records.append(Record(first, last))

for i in xrange(100):
  y = Records[:]
  y = sorted(y)
  #for w in y:
  #  print w.first, w.last

File: utils/benchmark/Phonebook/Phonebook.swift
@asmname("mach_absolute_time") func __mach_absolute_time__() -> UInt64

var words=[
"James", "John", "Robert", "Michael", "William", "David", "Richard", "Joseph",
"Charles", "Thomas", "Christopher", "Daniel", "Matthew", "Donald", "Anthony",
"Paul", "Mark", "George", "Steven", "Kenneth", "Andrew", "Edward", "Brian",
"Joshua", "Kevin", "Ronald", "Timothy", "Jason", "Jeffrey", "Gary", "Ryan",
"Nicholas", "Eric", "Stephen", "Jacob", "Larry", "Frank", "Jonathan", "Scott",
"Justin", "Raymond", "Brandon", "Gregory", "Samuel", "Patrick", "Benjamin",
"Jack", "Dennis", "Jerry", "Alexander", "Tyler", "Douglas", "Henry", "Peter",
"Walter", "Aaron", "Jose", "Adam", "Harold", "Zachary", "Nathan", "Carl",
"Kyle", "Arthur", "Gerald", "Lawrence", "Roger", "Albert", "Keith", "Jeremy",
"Terry", "Joe", "Sean", "Willie", "Jesse", "Ralph", "Billy", "Austin", "Bruce",
"Christian", "Roy", "Bryan", "Eugene", "Louis", "Harry", "Wayne", "Ethan",
"Jordan", "Russell", "Alan", "Philip", "Randy", "Juan", "Howard", "Vincent",
"Bobby", "Dylan", "Johnny", "Phillip", "Craig"]

// This is a phone book record.
struct Record : Comparable {
  var first : String
  var last : String

  init(_ first_ : String,_ last_ : String) {
    first = first_
    last = last_
  }
}
func ==(lhs: Record, rhs: Record) -> Bool {
  return lhs.last.compare(rhs.last) == 0 &&
         lhs.first.compare(rhs.first) == 0
}
func <(lhs: Record, rhs: Record) -> Bool {
  let lastComp = lhs.last.compare(rhs.last)
  if lastComp == -1 {
    return true
  }
  if lastComp == 1 {
    return false
  }

  let firstComp = lhs.first.compare(rhs.first)
  if firstComp == -1 {
    return true
  }

  return false
}

// The list of names in the phonebook.
var Names : Record[] = []
for first in words {
  for last in words {
    Names.append(Record(first, last))
  }
}

func benchStringSort() {
  let start = __mach_absolute_time__()
  for (var i = 0; i < 100; i++) {
    var t = Names;
    sort(&t)
  }
  let delta = __mach_absolute_time__() - start

  print("\(delta) nanoseconds.")
}

benchStringSort()
